web/plugs/authorization.ex

```diff
@@ -19,27 +19,51 @@ defmodule Cog.Plug.Authorization do
       raise ":permission key must be a string, but was '#{inspect permission}' instead"
     end
     {_,_} = Permission.split_name(permission) # error if can't be split
-    permission
+    opts
   end
 
   # Expects to be called after the `Cog.Plug.Authentication` plug
-  def call(conn, permission_name) do
+  def call(conn, opts) do
+    permission_name = Keyword.fetch!(opts, :permission)
     authenticated_user = get_user(conn)
     permission = name_to_permission(permission_name)
     case User.has_permission(authenticated_user, permission) do
       true ->
         conn
       false ->
-        conn
-        |> put_status(:forbidden)
-        |> json(%{error: "Not authorized."})
-        |> halt
+        if Keyword.get(opts, :allow_self_updates, false) do
```

Comment:
Adding a blurb about `:allow_self_updates` in the docstring for `init/1` would be helpful.